### ISSUE TITLE
server doesn't enforce internally-consistent settings

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri#7281]
### Observed behavior

Certain values of facility and device settings are incompatible with each other. For example. you cannot have:

```
learner_can_login_with_no_password: true
learner_can_edit_password: true
```

because learners cannot edit their password when they don't have a password.

This behavior is enforced on the client-side but not the server-side, e.g. during device provisioning.

### Expected behavior

Server-side model validation, checked by unit tests, should ensure that server state is always internally consistent.

### Context

0.14.0 beta 8

**Comment by ashmeet13** [comment]
Hi, can I pick this up? 
From what I understand going through the code - this [model class](https://github.com/learningequality/kolibri/blob/c655c7dcb03064d5764d7066a52999699c01178b/kolibri/core/auth/models.py#L98-L130) requires a clean() function that would validate such case?

New to Kolibri. Would really like if someone could guide me!

**Comment by jonboiser** [comment]
@ashmeet13 I think implementing the `clean` method would work for the example in the original post. You might also need to modify some code on the API layer to call `clean` before saving. For example, this code here for `PATCHING` the setting does not do a  `clean` before `save`.

https://github.com/learningequality/kolibri/blob/10733c9a1fb0fba9122411e5a259d7e7fa0b82e8/kolibri/core/device/api.py#L151

I think we still need to figure out the full set of "inconsistent" settings, but you're free to get started with the one in the original post where a `DeviceSettings` model is in an invalid state if both
```
learner_can_login_with_no_password: true
learner_can_edit_password: true
```

fields are true.

**Comment by ashmeet13** [comment]
Got it - I think for now I will start implementing the `clean()` function for

[kolibri/core/auth/models.py - FacilityDataset Model](https://github.com/learningequality/kolibri/blob/release-v0.14.x/kolibri/core/auth/models.py#L98)

This should solve the problem for the case in the original post.

**Comment by rtibbles** [comment]
Fixed in #7856
